source/contribute/maintainer.rst

```diff
@@ -59,16 +67,167 @@ Optionally, add any additional contributor/maintainer forks, e.g.:
+   behavior that was erroneously introduced, and prevents installation
+   or usage of the library by a very large portion of the community.
+-  **security fixes** are changes to the codebase that correct existing
+   behavior in the codebase that may lead to substantial financial or
+   personal damage to consumers of the packages due to malicious
+   attack vectors. Read about our `Security Policy <https://www.doctrine-project.org/policies/security.html>`_
+-  **deprecations** are changes to the codebase that do mark existing
+   API as "to be removed in future". Read about our `Deprecation Policy <https://www.doctrine-project.org/policies/deprecation.html>`_
+-  **backwards-compatibility breakages** (or **BC breaks** in short)
+   are modifications to the existing API or implementation that would
+   result in downstream users having to correct their software to
+   adapt to the new changes. Maintainers should also prevent any
+   unnecessary BC breaks, and always evaluating if it is worth
+   introducing them.
+
+Stability and Semantic Versioning
+---------------------------------
+
+A maintainer must also always consider that any proposal, regardless
+how well tested and verified it could be, brings in some instability.
+In order to reduce the amount of defects and/or regressions reaching
+downstream users, a maintainer must therefore always consider
+carefully where a patch may land.
+
+Packages controlled by the Doctrine organisation are to follow
+`Semantic Versioning (SemVer) <https://semver.org/spec/v2.0.0.html>`_
+rules, with the famous ``MAJOR.MINOR.PATCH`` naming scheme.
+
+This effectively means:
+
+- **PATCH** only contains **bug fixes**, **security fixes** and
+  **critical fixes**
+- **MINOR** can contain everything that is in **PATCH** plus
+  **new features**, **improvements** and **deprecations**
+- **MAJOR** can contain all of the above plus **BC breaks**
+
+Whilst it is possible to automate some of these decisions, humans
+are still better at categorising these changes due to the amount of
+nuances that are involved in the software development process.
+
 Branching Model
 ---------------
 
-Merging topic branches:
+In order to maintain all the stability invariants that SemVer imposes,
+it is vital that maintainers know where to merge incoming patches.
+
+Packages in the doctrine organisation should use following branching
+structure:
+
+ * ``develop`` - extremely unstable, points at the next planned
+   **MAJOR** release, may be rebased in order to speed up individual
+   maintainers prototyping new changes. Changes on ``develop`` can
+   be radical, and should not be relied upon.
+ * ``master`` - always to be considered as the next planned **MAJOR**
+   or **MINOR** release (depending on team internal agreement).
+   Consumers should not rely on ``master`` unless they are prepared
+   to adapt their codebase at every potentially breaking change.
+ * ``MAJOR.MINOR.x`` - always to be considered the next planned
```

Comment:
Considering that we currently mostly use `MAJOR.MINOR`, what is the motivation to use `MAJOR.MINOR.x`?

Reply:
ORM, DBAL and common started using `MAJOR.MINOR` with 2.2, while the rest either doesn't have enough branches for it to be called a model or are using `MAJOR.MINOR.x`.

Reply:
Fwiw either branch name format works with the website.